$OpenBSD: patch-src_gtkSignal_ml4,v 1.1 2013/02/01 10:22:01 dcoppa Exp $

[Bugfix] GtkSignal.connect_property: don't fetch the property value
by hand (upstream git commit adf6b7a5972356eddfdd6cb5243e4eaf8ea59d02)

--- src/gtkSignal.ml4.orig	Thu Aug 23 12:37:48 2012
+++ src/gtkSignal.ml4	Fri Feb  1 09:46:33 2013
@@ -200,6 +200,6 @@ let connect ~sgn: ((sgn:('a, _) t)) ~callback ?after (
 
 let connect_property ~(prop:('a, _) property) ~callback (obj : 'a obj) =
   let name = "notify::" ^ prop.Gobject.name in
-  let callback = fun () -> callback (get prop obj) in
-  connect_aux ~name ~marshaller:marshal_unit ~callback obj
+  let marshaller = marshal1 prop.conv name in
+  connect_aux ~name ~marshaller ~callback obj
 
